lib-multisrc/greenshit/src/eu/kanade/tachiyomi/multisrc/greenshit/GreenShitDto.kt

```diff
@@ -1,240 +1,259 @@
+    @JsonNames("cap_numero", "number")
+    val numero: Float? = null,
+    @JsonNames("cap_lancado_em", "cap_liberar_em", "cap_criado_em")
+    val releaseDate: String? = null,
+) {
+    fun toSChapter() = SChapter.create().apply {
+        this.name = this@ChapterDto.name
+        url = "/capitulo/$id"
+        numero?.let { chapter_number = it }
+        date_upload = ResultDto.DATE_FORMAT.tryParse(releaseDate)
+    }
+}
+
+@Serializable
+class FiltersDto(
+    @SerialName("generos") val genres: List<Genre> = emptyList(),
+    @SerialName("formatos") val formats: List<Format> = emptyList(),
+    @SerialName("status") val statuses: List<Status> = emptyList(),
+    @SerialName("tags") val tags: List<Tag> = emptyList(),
 )
 
 @Serializable
-class WrapperChapterDto(
-    @SerialName("capitulos")
-    val chapters: List<ChapterDto>,
+class Genre(
+    @JsonNames("gen_id", "id")
+    val id: Int,
+    @JsonNames("gen_nome", "name")
+    val name: String,
 )
 
 @Serializable
-class ChapterPageDto(
-    @SerialName("cap_paginas")
-    val pages: List<PageDto>,
-    @SerialName("obra")
-    val manga: MangaReferenceDto,
-    @SerialName("cap_numero")
-    val chapterNumber: Float,
+class Format(
+    @JsonNames("formt_id", "id")
+    val id: Int,
+    @JsonNames("formt_nome", "name")
+    val name: String,
+)
+
+@Serializable
+class Status(
+    @JsonNames("stt_id", "id")
+    val id: Int,
+    @JsonNames("stt_nome", "name")
+    val name: String,
+)
+
+@Serializable
+class Tag(
+    @JsonNames("tag_id", "id")
+    val id: Int,
+    @JsonNames("tag_nome", "name")
+    val name: String,
+)
+
+@Serializable
+class ChapterPagesDto(
+    @SerialName("obr_id") val mangaId: Int? = null,
+    @SerialName("cap_nome") val name: String = "",
+    @SerialName("cap_numero") private val _chapterNumber: JsonPrimitive? = null,
+    @SerialName("cap_paginas") val pages: List<PageDto> = emptyList(),
+    @SerialName("cap_texto") val text: String? = null,
+    @SerialName("cap_tipo") val type: String = "IMAGEM",
 ) {
-    @Serializable
-    class MangaReferenceDto(
-        @SerialName("obr_id")
-        val id: Int,
-        @SerialName("scan_id")
-        val scanId: Int,
-    )
+    val chapterNumber: String? get() = _chapterNumber?.content
+
+    fun toPageList(cdnUrl: String): List<Page> {
+        if (type == "TEXTO") {
+            throw IllegalArgumentException("Novels are not supported")
+        }
+
+        return pages.mapIndexedNotNull { index, pageDto ->
+            pageDto.toPageOrNull(cdnUrl, mangaId, chapterNumber)?.let { Page(index, it.url, it.imageUrl) }
+        }
+    }
 }
 
 @Serializable
 class PageDto(
     val src: String,
-    @SerialName("numero")
-    val number: Float? = null,
+    val mime: String? = null,
+    var path: String? = null,
+    @SerialName("numero") private val _numero: JsonPrimitive? = null,
 ) {
-    fun isWordPressContent(): Boolean = number == null
+    val numero: String? get() = _numero?.content
```

Comment:
I think you can use the field directly with:
```suggestion
    @Serialize(with = SomeCustomSeralizer::class)
    val numero: String? = null,
) {
```

Might be more work than what it's worth.